### ISSUE TITLE
PowerShell beta hangs in Bash on Windows shell

### ISSUE THREAD

**Issue by jdhitsolutions** [PowerShell/PowerShell#3799]
Before the latest beta I was able to install and run the PowerShell 6 alphas in the Bash shell on Windows 10. Now, with the beta install, the shell hangs when I try to start PowerShell.  I realize I could post an issue on GitHub but thought I'd check with the group first.

I am running Windows 10 build 16193.
In the Bash shell, which I just upgraded as well, I can see that PowerShell beta is installed.
```
powershell/xenial,now 6.0.0-beta.1-1ubuntu1.16.04.1 amd64 [installed]
```
lsb_release -a shows:
```
Distributor ID: Ubuntu
Description:    Ubuntu 16.04.2 LTS
Release:        16.04
Codename:       xenial
```
I've even tried running powershell -noprofile with no change.

Ideas or suggestions for things to look at? 

I see a similar comment at the end of #3747

**Comment by bgelens** [comment]
The same happens with me. 
Looks similar to the issue with vs code PS ext. https://github.com/PowerShell/vscode-powershell/issues/742

**Comment by jdhitsolutions** [comment]
I wondered as well if the two issues aren't related.

**Comment by jdhitsolutions** [comment]
Because I had nothing left to try, I created profile scripts and put in the [System.Console]::OutputEncoding = [System.Text.Encoding]::ASCII line from the VSCode issue. Didn't make any difference.

**Comment by jdhitsolutions** [comment]
This is interesting. I was able to get something like this to work in the Bash shell.
```
powershell -nologo -OutputFormat Text -command "&{Get-Process}"
```

**Comment by MaximoTrinidad** [comment]
I Already logged the incident in WSL GitHub awhile ago: PowerShell 6.0.0-beta hung up in Windows 10 Bash Console. #2124 

:)

**Comment by MaximoTrinidad** [comment]
@jdhitsolutions,

Your line works!  I tested the PowerShell Beta before installing .NET Code 2.0 Preview and it was working, meaning that I can get to the prompt.  I was still having issues typing in the console with PowerShell Beta in the prompt.

:)

**Comment by MaximoTrinidad** [comment]
But, I can still use PowerShell in Windows 10 Bash in Ubuntu Desktop. 

![poshbeta_bashgui_2017-05-17_10-28-40](https://cloud.githubusercontent.com/assets/19892435/26170150/615b49fe-3b0e-11e7-91b3-75dfb1408a33.jpg)

**Comment by jdhitsolutions** [comment]
I can see what I'm supposed to be running.
```
jeff@YPJH10:/home$ powershell -nologo -command "&{Get-Variable PSVersionTable | Select -expand Value}"

Name                           Value
----                           -----
PSVersion                      6.0.0-beta
PSEdition                      Core
BuildVersion                   3.0.0.0
CLRVersion
GitCommitId                    v6.0.0-beta.1
OS                             Linux 4.4.0-43-Microsoft #1-Microsoft Wed Dec 31 14:42:53 PST 2014
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by MaximoTrinidad** [comment]
But!...  You can't get to the prompt.  Does by any chance, have you installed .NET Core in Bash?
:)

**Comment by MaximoTrinidad** [comment]
I got the notice on WSL "Microsoft .net Core team and powershell team have active bugs listed on this github repo."

:)

**Comment by jdhitsolutions** [comment]
Not sure what version of .NET core is running in my Bash shell. I'm going through the .NET Core 2.0 Preview 1 setup now.

**Comment by MaximoTrinidad** [comment]
@jdhitsolutions,

I can tell you, installing .NET Core 2.0 Preview, that won't work.  It  was working before installing.

**Comment by MaximoTrinidad** [comment]
I'm more tempted to do a clean Bash setup, then do a PowerShell Beta install to see if the .NET Core 2.0 Preview had something to do with PowerShell not opening.  

But, I will have to do this on my VM at home after work this evening.

**Comment by MaximoTrinidad** [comment]
FYI
I completed testing PowerShell 6.0.0-beta on my Window 10 Bash Console. I uninstall and did a clean Bash installation without .NET Core 2.0 Preview 1 and PowerShell is getting hung.

1. lxrun /uninstall /full
2. lxrun /install
3. Open Bash Console.
4. sudo su ##-> Prepare for PowerShell Beta Install
    curl https://packages.microsoft.com/keys/microsoft.asc | sudo apt-key add -
    curl https://packages.microsoft.com/config/ubuntu/16.04/prod.list | sudo tee /etc/apt/sources.list.d/microsoft.list
     exit
5. sudo apt update
6. sudo apt upgrade ##-> This command will also install PowerShell
7. Open PowerShell ##-> But it hungs, the prompt won't come up.

![powershellhungbash](https://cloud.githubusercontent.com/assets/19892435/26181406/cf872b30-3b3d-11e7-800b-e1df14e3d0d7.PNG)

**Comment by MaximoTrinidad** [comment]
@daviwil !!

I did follow part of your instructions to see if I could bypass the issue with PowerShell in Bash creating the "profile.ps1" on $pshome.  I got it to echo me back but still the prompt won't show up.

profile.ps1
[System.Console]::OutputEncoding = [System.Text.Encoding]::ASCII
Write-Host "PowerShell on Bash 6.0.0-beta.1 loaded"

Any ideas!! So, freaking close. 
:)

![bashposhissue_2017-05-18_21-17-01](https://cloud.githubusercontent.com/assets/19892435/26265354/d5074b22-3cae-11e7-85a5-c4f0aeeb6ba0.jpg)

**Comment by daviwil** [comment]
That one I'm not certain about, unfortunately.  Hopefully they'll fix these issues with an upcoming Fast Ring build.  Let me check and see if I have a new Canary build to try with...

**Comment by MaximoTrinidad** [comment]
Thanks @daviwil!  At least the VS Code workaround was nice!!  
I've been using XServer and Ubuntu Desktop in Windows 10 which hasn't been that bad to work with PowerShell on Linux.

Greatly Appreciated,
:)

**Comment by daviwil** [comment]
Looks like I have access to 16202, will spin it up and give it a shot :)

**Comment by MaximoTrinidad** [comment]
Awesome @daviwil! I'll keep my finger crossed!! 
:)

**Comment by daviwil** [comment]
Same problem in 16202 :(

**Comment by MaximoTrinidad** [comment]
Well,  at least I'm not on a total lost as I can use Ubuntu Desktop Terminal to run PowerShell without any issues.  I not it will come around and be fixed later.

Thanks again,
:)

**Comment by jdhitsolutions** [comment]
This still appears to be an issue with beta 2. I can run commands like this 
```
powershell -nologo -command "&{Get-process | sort-object WS -desc | select -first 5}"
```
but no interactive session.

**Comment by jdhitsolutions** [comment]
Not sure if this will clarify things or make it more confusing. But on a clean Windows 10 build 14393 I installed WSL and then PowerShell beta 2. It works as expected. It does not work on Windows build 16199. At least for me.

**Comment by SteveL-MSFT** [comment]
Both the Windows console and WSL is changing, so it's going to be hard to determine what is actually causing the issue here.  We should revisit this when the console and WSL stabilize.

**Comment by jdhitsolutions** [comment]
I feel everything is a moving target these days. For what it's worth, on the older Win10 platforms WSL installs Ubuntu 14.04.05 LTS but in the latest builds it installs 16.04.2 LTS.

**Comment by jdhitsolutions** [comment]
Here's an update: Just installed a new Windows 10 build (16215). Thought I'd try this again, and it worked! Now, I also updated Ubuntu before I tried and it looks like I got some libc stuff so I don't know what made a difference. But I will say, that PSReadline caused issues typing. Cursor would jump back and forth on the line, sometimes it seemed even prematurely inserting a return,and in general caused a distracting typing experience.
